src/UI/Components/Diagram/shapes.ts

```diff
@@ -277,17 +301,42 @@ export class ServiceEntityBlock extends shapes.standard.HeaderedRecord {
     return wasThereRelationToRemove;
   }
 
+  /**
+   * Retrieves the name of the entity.
+   *
+   * @returns {string} The name of the entity.
+   */
   getName(): string {
     return this.get("entityName");
   }
 
-  setTabColor(color: string) {
-    const currentClassName = this.attr(["header", "class"]);
-
-    return this.attr(["header", "class"], `${currentClassName} -${color}`);
+  /**
+   * Sets the tab color based on the entity type.
+   *
+   * @param {EntityType} type - The type of the entity.
+   * @returns {this} updated entity block
+   */
+  setTabColor(type: EntityType): this {
```

Comment:
Why do you need to return "this" when setting an attribute? 🤔 
I think it would be valuable to have the colors in a type, which reflects the EntityType

Reply:
It's the convention that they had on the demo that the actions from the event return updated entity, but I can remove that, as I don't think we are using it

Reply:
It was just a question, maybe just add it in the comment.

Reply:
The color codes would have more meaning being a Type reflecting the EntityTypes
Also, don't forget to add the comment about "this"

Reply:
Added to general description of the shape

Reply:
I was only refering to this specific `setTabColor`. I get why in other methods you do : return `this.get(blabla)`. What is odd here in this specific method, is that you returnonly  the setting of the attribute, or the entire scope. 
It's not about just following JointJs examples in general. It must have a reason why we need the full scope in this case.

Can you also assign the Hex codes to an enum so they make more sense? I also wonder if we can't use the JS-tokens from PF (not the css variables, but their JS value counterparts) to avoid needing to update or hunt down wherever we use those color codes.